signin/index.js

```diff
@@ -60,55 +79,72 @@ function isValidEmail() {
   return true;
 }
 
-function isValidPassword() {
+//비밀번호 에러 확인
+function validatePassword() {
   const password = passwordInput.value;
-  const isPasswordEmpty = isEmpty(password);
   removeError(pwErrorMessage);
-  if (isPasswordEmpty) {
+  if (isEmpty(password)) {
     toggleError(pwErrorMessage, "비밀번호를 입력해 주세요.");
     return false;
-  } else {
-    removeError(pwErrorMessage);
-    return true;
   }
+  removeError(pwErrorMessage);
+  return true;
 }
 
-function checkEmailInput() {
-  toggleError(emailErrorMessage, "이메일을 확인해 주세요.");
-}
+//이메일, 비밀번호가 회원 정보와 일치하는지 확인
+async function isValidUser(loginInfo) {
+  try {
+    const response = await fetch("https://bootcamp-api.codeit.kr/api/sign-in", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(loginInfo),
+    });
 
-function checkPasswordInput() {
-  toggleError(pwErrorMessage, "비밀번호를 확인해 주세요.");
-}
+    if (!response.ok) {
+      // throw new Error(response.statusText);
+      // const errorResponse = await response.json();
+      // const errorMessage = errorResponse.error.message;
+      // throw new Error(errorMessage);
+      //error을 던지면 서버 오류와 같이 처리되기 때문에 그냥 false를 리턴
+      return false;
+    }
 
-function isValidUser(loginInfo) {
-  const validUserSchema = { email: "test@codeit.com", password: "codeit101" };
-  const { email, password } = loginInfo;
-  if (email !== validUserSchema.email) {
-    return false;
-  }
-  if (password !== validUserSchema.password) {
-    return false;
+    const result = await response.json();
+    const accessToken = result.data.accessToken;
+    localStorage.setItem("accessToken", accessToken);
+    location.href = "../folder/index.html";
+    return true;
+  } catch (err) {
+    alert("서버 오류가 발생했습니다.");
   }
-  location.href = "../folder/index.html";
-  return true;
 }
```

Comment:
@q45402sk 



**[👍🏻 ]**
```suggestion
 if (!response.ok) {
      // throw new Error(response.statusText);
      // const errorResponse = await response.json();
      // const errorMessage = errorResponse.error.message;
      // throw new Error(errorMessage);
      //error을 던지면 서버 오류와 같이 처리되기 때문에 그냥 false를 리턴
      return false;
    }
}
```
- 이 부분 너무 좋아요! 
- 네트워크 요청하는 서비스 로직에서 에러핸들링을 굳이 하지 않고, 이 함수가 실제 호출되는곳에서 에러핸들링을 하면 사용자에게 적절한 UI 업데이트를 제공해줄 수 있겠죠 ㅎㅎ 자세한건 아래 제안사항 참고해주세요

**[제안]**
- 이제 서비스 로직과 비즈니스 로직이 섞이게 돼죠 ㅎㅎ
- 이처럼 네트워크에 요청해 데이터를 가져오는 서비스 로직의 경우, 실제 로그인을 위한 네트워크 요청인 만큼 함수 이름고 ㅏ하는 역할의 분리가 있으면 좋을것 같습니다. 
```js


async function loginUser(loginInfo) {
  const { email, password } = loginInfo

  const response = await fetch("login url")

  // 여기서 네트워크 오류 발생 여부를 체크합니다.
  // 참고: https://developer.mozilla.org/en-US/docs/Web/API/Response/ok
  if (!response.ok) {
    const errorDetail = await response.text()
    throw new Error(`서버 오류가 발생했습니다: ${errorDetail}`)
  }

  const json = await response.json()
  const { accessToken, user } = json

  return { user, accessToken }
}

```